### ISSUE TITLE
Unable to loop across object properties after using `defineProperty`

### ISSUE THREAD

**Issue by 0x0049** [NeilFraser/JS-Interpreter#136]
My particular use case is trying to use the interpreter on some code bundled by Webpack. It uses `defineProperty` to set a property called `__esModules` and then later loops across that object and assigns it to an external `exports` object.

```
// Setting `copyFrom.test` *before* defineProperty (or removing defineProperty altogether)
// makes it work.
const code = `
	var copyFrom = {}, copyTo = {};
	Object.defineProperty(copyFrom, "test2", { value: "test2" });
	copyFrom.test = "test";

	console.log("Should be undefined:", copyTo.test);
	for (var i in copyFrom) {
		console.log("copying:", i);
		copyTo[i] = copyFrom[i];
	}
	console.log("Should be \\"test\\":", copyTo.test);
`;

eval(code); // Works as expected.

console.log("----------------");

const interpreter = new Interpreter(code, (intp, scope) => {
	intp.setProperty(scope, "console", intp.nativeToPseudo({
		log: (...args) => console.log(...args),
	}));
});
interpreter.run();

/*
	Output:
	Should be undefined: undefined
	copying: test
	Should be "test": test
	----------------
	Should be undefined: undefined
	copying: constructor
	Should be "test": undefined
*/
```

**Comment by NeilFraser** [comment]
Really interesting bug.  Looks like the for-in loop is breaking the first time it sees a non-enumerable property.  Around line 3026.  I'll look into this further in a few hours...

**Comment by NeilFraser** [comment]
Fixed in a649f8149af04dcd46279e67f1b4867a9cf508b3
I'll recompile and update the demos tomorrow.

**Comment by NeilFraser** [comment]
Recompiled and pushed.  Thanks for the bug!
